app/presenters/event_instance_presenter.rb

```diff
@@ -55,18 +55,27 @@ def video_embed_link
   private
 
   def map_to_users(participants)
-    participants ||= []
-
-    participants.map do |participant|
-      begin
-        person = participant.last['person']
-        user = Authentication.find_by(provider: 'gplus', uid: person['id']).try!(:user)
-        next if user == host
-        user || NullUser.new(person[:displayName])
-      rescue NoMethodError
-        Rails.logger.error "Exception at event_instance_presenter#map_to_users"
-        nil
-      end
-    end.compact
+    participants ||= ActionController::Parameters.new({})
+    return participants_as_array(participants) if participants.is_a?(Array)
+    participants_as_parameters(participants)
+  end
+
+  def participants_as_parameters(participants)
+    participants.to_unsafe_hash.map{ |participant| process_users(participant) }.compact
+  end
+
+  def participants_as_array(participants)
+    participants.map{ |participant| process_users(participant) }.compact
+  end
+
```

Comment:
@tansaku I would like to unify those two methods above, I tried a couple of things but none of them convinced me. Do you think this way is ok, or should I change it?

Reply:
I would probably hold off refactoring here without better understanding the context of these methods

Reply:
This was not done in order to refactor, you will understand better if you pull the changes and run it locally. You will find that this method is receiving `participants` sometimes as an `Array` and sometimes as `ActionController::Parameters` object, so in order to handle both I had to split the code.

Reply:
aha, then maybe:

```ruby 
  def map_to_users(participants)
    participants ||= ActionController::Parameters.new({})
    participants = participants.to_unsafe_h unless participants.is_a?(Array)
    participants.map{ |participant| process_users(participant) }.compact
  end
```